common/models/stock-order-lineitem-model.js

```diff
@@ -382,5 +382,150 @@ module.exports = function(StockOrderLineitemModel) {
         });
     }
   };
-
+    StockOrderLineitemModel.scanBarCode = function (scanType, productSku, orgModelId, reportModelId, force) {
+        logger.debug({
+            functionName: 'scanBarCode',
+            message: 'Will find all stock line items and increment quantity',
+            scanType,
+            productSku,
+            orgModelId,
+            reportModelId,
+            force
+        });
+        const filter = {
+            where: {
+                orgModelId,
+                reportModelId,
+                approved: true,
+            },
+            include: {
+                relation: 'productModel',
+                scope: {
+                    where: {
+                        sku: productSku
+                    }
+                }
+            }
+        };
+        if (scanType === 'receive'){
+            filter.where.fulfilled = true;
+        }
+           return StockOrderLineitemModel.find(filter).then(function (orderLineItems) {
+               // Remove all objects where productModel is not found
+                orderLineItems = orderLineItems.filter(function (lineItem) {
```

Comment:
write product search first, then search for that product in line items. Will be faster than this filter.